File: data/java/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER_TRANSLATED.py
====================
def f_gold ( arr , n ) :
    q = Queue ( )
    arr.sort ( )
    q.put ( arr [ 0 ] )
    for i in range ( 1 , n ) :
        now = q.get ( )
        if arr [ i ] >= 2 * now :
            q.put ( )
        q.put ( arr [ i ] )
    return q.get ( )

File: data/python/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER_TRANSLATED.java
====================
public static final int fGold ( int [ ] arr , int n ) {
  Queue < Integer > q = new LinkedList < Integer > ( ) ;
  Arrays . sort ( arr ) ;
  q . add ( arr [ 0 ] ) ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    int now = q . poll ( ) ;
    if ( ( arr [ i ] >= 2 * now ) && ( arr [ i ] <= 2 * now ) ) q . remove ( i ) ;
    q . add ( arr [ i ] ) ;
  }
  return q . size ( ) ;
}

